### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by lpinner** [scopatz/xonsh#930]
Running `xonfig wizard` and trying to set `$HISTCONTROL` raises an exception.

```
<snip>
$HISTCONTROL
A set of strings (comma-separated list in string form) of options
that determine what commands are saved to the history list. By
default all commands are saved. The option 'ignoredups' will not save
the command if it matches the previous command. The option
'ignoreerr' will cause any commands that fail (i.e. return non-zero
exit status) to not be added to the history list.
default value: set()
current value: {'ignoreboth'}
>>> ignoredups
Converted input 'ignoredups' to {'ignoredups'}.
<snip>
TypeError: {'ignoredups'} is not JSON serializable
```
$ xonsh --version
('xonsh/0.2.7',)

Installed from pip.

**Comment by scopatz** [comment]
Hi @lpinner - this is definitely a bug... Can you please send us the full traceback, though. Thanks!

**Comment by lpinner** [comment]
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/xonsh/xonfig.py", line 219, in _wizard
    pv.visit()
  File "/usr/local/lib/python3.5/dist-packages/xonsh/wizard.py", line 469, in visit
    rtn = super().visit(node)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/wizard.py", line 292, in visit
    rtn = meth(node)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/wizard.py", line 520, in visit_wizard
    self.visit(child)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/wizard.py", line 469, in visit
    rtn = super().visit(node)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/wizard.py", line 292, in visit
    rtn = meth(node)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/wizard.py", line 577, in visit_save
    jstate = json.dumps(self.state, indent=1, sort_keys=True)
  File "/usr/lib/python3.5/json/__init__.py", line 237, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.5/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib/python3.5/json/encoder.py", line 429, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.5/json/encoder.py", line 403, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.5/json/encoder.py", line 403, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.5/json/encoder.py", line 436, in _iterencode
    o = _default(o)
  File "/usr/lib/python3.5/json/encoder.py", line 180, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: {'ignoredups'} is not JSON serializable

```

**Comment by scopatz** [comment]
Thanks!

**Comment by scopatz** [comment]
Hi @lpinner - I have put in a potential fix to #974

**Comment by melund** [comment]
I just confirmed that it fixed the problem on my system.
